assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -231,6 +239,21 @@ class AssistedInjectProcessor : AbstractProcessor() {
   }
 }
 
+internal data class AssistedInjectRequest(
+  val type: TypeElement,
+  val factoryType: TypeElement,
+  val factoryMethod: ExecutableElement,
+  val parameterKeys: List<ParameterKey>
+) {
+  fun toInjection(): AssistedInjection {
+    val targetType = TypeName.get(type.asType())
+    val factoryType = ClassName.get(factoryType)
+    val returnType = ClassName.get(factoryMethod.returnType)
+    val methodName = factoryMethod.simpleName.toString()
+    return AssistedInjection(targetType, parameterKeys, factoryType, methodName, returnType)
```

Comment:
use parameter naming instead of all these fields?

Reply:
I can do that on the latest commit. There's a bunch between me and you... I was trying to break this up into reviewable bits.